homeassistant/components/mqtt/climate.py

```diff
@@ -405,9 +406,15 @@ def render_template(msg, template_name):
         def handle_action_received(msg):
             """Handle receiving action via MQTT."""
             payload = render_template(msg, CONF_ACTION_TEMPLATE)
-
-            self._action = payload
-            self.async_write_ha_state()
+            if payload in CURRENT_HVAC_ACTIONS:
+                self._action = payload
+                self.async_write_ha_state()
+            else:
+                _LOGGER.error(
```

Comment:
I suggest we log a warning here, not an error.

Reply:
I agree with you - warning should better than error.

Reply:
Cheers, adjusted it accordingly